Automattic/newspack-newsletters
# feat: ad placeholder block

### All Submissions:






### Changes proposed in this Pull Request:

Implements the "Newsletter Ad" block for manual selection and placement of ads in the newsletter:

<img width="1207" alt="image" src="https://github.com/Automattic/newspack-newsletters/assets/820752/d95d1e96-729c-4973-bd36-5e2cba580742">

<img width="282" alt="image" src="https://github.com/Automattic/newspack-newsletters/assets/820752/d9d8b08e-d659-4a2e-b0ef-a92739056367">


For the "automatic selection", the selection is based on the position percentage. The ads are sorted by position and, regardless of the position the block is placed, the first positioned block will render the first position percentage.

### How to test the changes in this Pull Request:

1. Check out this branch and make sure you have active newsletter ads (at least 2)
2. Draft a new newsletter and place a "Newsletter Ad" block
3. Confirm it renders with "Automatic selection" as default
4. Place another "Newsletter Ad" and click to Send
5. Confirm the preview renders both ads in the correct order (according to their position)
6. Close the modal and select the ad for each block (instead of automatic selection)
7. Confirm the choices are reflected when clicking to Send
8. Also confirm the newsletter's. "Ads Settings" shows the automatic insertion of ads disabled with proper notice, as in the image above

### Other information:
